test/test_nn.py

```diff
@@ -3317,49 +3323,76 @@ def _multihead_attn_test_helper(use_src_lengths):
                 K = np.random.rand(*dims).astype(np.float64)
                 V = K
                 Q = np.expand_dims(decoder_state, 1)
+                attn_mask = np.random.randint(0 , 2, size=(1, seq_len))
```

Comment:
Don't rely on numpy, you need to assume it's not available. If you absolutely need it, let's sync up and we can add a decorator to make an exception, but it's not recommended because it makes it less likely for your test to run.

Reply:
several functions in the test_multihead_attention test were re-writed in numpy version, like softmax and batchmatmul.